File: backend/internal/model/dto/tenant.go
package dto

import (
	"github.com/linbaozhong/gentity/pkg/types"
	"time"
)

// ============ Tenant 请求 ============

// RegisterTenantReq 租户自助注册请求
// @request
type RegisterTenantReq struct {
	Name         *string `json:"name" valid:"required"`          // 企业名称
	Logo         *string `json:"logo"`                           // 企业logo
	ContactName  *string `json:"contact_name" valid:"required"`  // 联系人姓名
	ContactPhone *string `json:"contact_phone" valid:"required"` // 联系电话（即管理员手机号）
	Password     *string `json:"password" valid:"required"`      // 管理员密码
	WxCode       *string `json:"wx_code"`                        // 微信登录code（可选，绑定微信）
}

// OpenTenantReq 平台开通/解锁租户请求
// @request
type OpenTenantReq struct {
	ID *int64 `json:"id" valid:"required"` // 租户编号
}

// SwitchTenantReq 切换当前租户请求
// @request
type SwitchTenantReq struct {
	TenantID *int64 `json:"tenant_id" valid:"required"` // 目标租户编号
}

// InviteMemberReq 邀请成员请求
// @request
type InviteMemberReq struct {
	TenantID *int64  `json:"tenant_id" valid:"required"` // 租户编号
	Phone    *string `json:"phone" valid:"required"`     // 被邀请人手机号
	RoleID   *int64  `json:"role_id" valid:"required"`   // 角色编号
}

// JoinMemberReq 加入已有企业（邀请码）请求
// @request
type JoinMemberReq struct {
	InviteCode *string `json:"invite_code" valid:"required"` // 邀请码
}

// ============ Tenant 响应 ============

// TenantResp 租户信息
// @response
type TenantResp struct {
	ID           types.BigInt `json:"id"`            // 租户编号
	Name         string       `json:"name"`          // 企业名称
	Logo         string       `json:"logo"`          // 企业logo
	ContactName  string       `json:"contact_name"`  // 联系人姓名
	ContactPhone string       `json:"contact_phone"` // 联系电话
	Plan         string       `json:"plan"`          // 套餐
	Status       int8         `json:"status"`        // 状态：0试用中 1正式 2锁定
	TrialStart   time.Time    `json:"trial_start"`   // 试用开始
	TrialEnd     time.Time    `json:"trial_end"`     // 试用结束
}

// UserTenantListResp 用户已绑定企业列表
// @response
type UserTenantListResp struct {
	Tenants []TenantResp `json:"tenants"` // 企业列表
	Current types.BigInt `json:"current"` // 当前企业编号（0表示未选择）
}

// MemberResp 成员信息
// @response
type MemberResp struct {
	ID       types.BigInt `json:"id"`        // 成员编号
	UserID   types.BigInt `json:"user_id"`   // 全局用户编号
	Name     string       `json:"name"`      // 姓名
	Phone    string       `json:"phone"`     // 手机号
	RoleID   types.BigInt `json:"role_id"`   // 角色编号
	Status   int8         `json:"status"`    // 状态：0在职 1离职
	AuthUrls []string     `json:"auth_urls"` // 权限url列表
}

File: mp-b/admin/api/tenant.ts
/**
 * 租户相关接口
 * 对应后端 TenantController（/tenant/...）
 */
import { get, post } from '../utils/request'
import { setLoginUser, LoginUser } from '../utils/auth'

export interface TenantResp {
	id: number
	name: string
	logo: string
	contact_name: string
	contact_phone: string
	plan: string
	status: number
	trial_start: string
	trial_end: string
}

export interface UserTenantList {
	tenants: TenantResp[]
	current: number
}

export interface RegisterParams {
	name: string
	logo?: string
	contact_name: string
	contact_phone: string
	password: string
	wx_code?: string
}

// ====== MOCK 数据（后台未完成，测试前端用；后台完成后放开各函数内注释并删除本段）======
// 模拟“多企业”场景：选择企业页会列出 2 家企业
const mockTenantList: TenantResp[] = [
	{ id: 1, name: '示例养老院', logo: '', contact_name: '张三', contact_phone: '13500000000', plan: '专业版', status: 1, trial_start: '2026-01-01', trial_end: '2027-01-01' },
	{ id: 2, name: '康泰护理中心', logo: '', contact_name: '李四', contact_phone: '13500000001', plan: '标准版', status: 1, trial_start: '2026-02-01', trial_end: '2027-02-01' }
]
const mockLoginUser: LoginUser = {
	id: 1,
	name: '管理员',
	avator: '',
	phone: '13800000000',
	tenant_id: 1,
	member_id: 1,
	role_id: 1,
	role_name: '管理员',
	need_bind: true,
	auth_id_list: [],
	auth_url_list: ['/people/old', '/check-in/leave', '/check-in/visit', '/check-in/accident', '/food/order', '/live/bed', '/live/enter', '/live/apply', '/serve/book', '/tenant/member', '/home/dashboard', '/audit'],
	tenants: mockTenantList,
	token: 'mock-token-001'
}

/** 租户自助注册：成功后写入登录态 */
export async function registerTenant(params: RegisterParams): Promise<LoginUser> {
	// const data = await post<LoginUser>('/tenant/register', params)
	// setLoginUser(data)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	const mockUser: LoginUser = { ...mockLoginUser, name: params.name, phone: params.contact_phone }
	setLoginUser(mockUser)
	return mockUser
}

/** 我的企业列表 */
export function myTenants(): Promise<UserTenantList> {
	// return get<UserTenantList>('/tenant/myTenants')
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve({ tenants: mockTenantList, current: 1 })
}

/** 切换当前租户：改本地登录态里的当前企业 id */
export async function switchTenant(tenantId: number): Promise<LoginUser> {
	// const data = await post<LoginUser>('/tenant/switchTenant', { tenant_id: tenantId })
	// setLoginUser(data)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	const mockUser: LoginUser = { ...mockLoginUser, tenant_id: tenantId }
	setLoginUser(mockUser)
	return mockUser
}

/** 加入已有企业 */
export async function joinMember(inviteCode: string): Promise<LoginUser> {
	// const data = await post<LoginUser>('/tenant/joinMember', { invite_code: inviteCode })
	// setLoginUser(data)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	setLoginUser(mockLoginUser)
	return mockLoginUser
}

/** 邀请成员 */
export function inviteMember(params: { tenant_id: number; phone: string; role_id: number }): Promise<void> {
	// return post<void>('/tenant/inviteMember', params)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}
